api/internal/localizer/locloader.go

```diff
@@ -0,0 +1,382 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package localizer
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"sigs.k8s.io/kustomize/api/ifc"
+	"sigs.k8s.io/kustomize/api/internal/git"
+	"sigs.k8s.io/kustomize/api/loader"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+const dstPrefix = "localized"
+const localizeDir = "localized-files"
+const fatalPrefix = "should never happen"
```

Comment:
"should never happen" is a weird thing to log in an error message, we probably shouldn't include this. You can leave it as a comment where the error is returned if you'd like.